### ISSUE TITLE
[QUESTION] Unable to double click to adjust the loops that I have already laid down

### ISSUE THREAD

**Issue by laptop-laptop** [CGCookie/retopoflow#1274]
**PLEASE READ THE FOLLOWING AND USE THE TEMPLATE TO POST YOUR ISSUE**

*Questions not directly related to the RetopoFlow project are considered general and should be posted elsewhere (e.g. git, blender, python, etc).*

**Before raising the issue, please check the following first:**

- [ ] Read through the README.md file in the branch used to assure the process taken is correct
- [ ] Check the existing [issues](https://github.com/CGCookie/retopoflow/issues) and [pull requests](https://github.com/CGCookie/retopoflow/pulls) to make sure the issue has not already been reported and/or fixed.
- [ ] You can replicate the bug and will provide as much info as possible with blend files, screenshots, logfiles, as well as show the expected result and actual result.

&nbsp;

Issue Template
-----------------------------------

---
Please prefix your issue name with one of the following: **[BUG]** **[PROPOSAL]** **[QUESTION]**

Retopoflow Version: 3.4.0

Blender Version/Hash: 3.6.2

Platform Version/Distribution: Windows 10

&nbsp;

**Issue:**

I used the tool Contours to create three loops surrounding the front leg of a horse model. From "The Complete Guide to Retopology with RetopoFlow 3 in Blender" video (10:37), I double clicked on a loop and then left clicked and dragged to slide it along the mesh. I have tried many times on each of the three loops but it worked only 1 or 2 times out of 10 attempts. Sometimes a vertical line got selected instead. What have I done wrong?

https://www.youtube.com/watch?v=9OjS2rU-Jr0&t=507s

https://imgur.com/a6Jlnr3

![image](https://github.com/CGCookie/retopoflow/assets/3790699/c7e7b65a-d8b4-4602-96a2-311cd62077e4)

Also, what is the difference between red and white loops?

Is it possible to add options to increase the thickness of lines and size of the dots, perhaps also change the color of these? It is a bit hard to see them clearly.

Don't know if this is related. As I continued to the next step in the video to create a non-uniform cut by going to Tweak to move the vertices, I was also unable to click, select and move those white dots.

(edit: added image to GH issue)

**Comment by vxlcoder** [comment]
I haven't looked into the issue you reported, yet, but below are some answers to your other questions.  note: i just made a few changes that should help improve readability of this geometry, and my answers below reflect those changes.  use the following screenshot as a reference.

![image](https://github.com/CGCookie/retopoflow/assets/3790699/1df06165-1dc7-40f0-9448-0c5e30d0f285)

white edges have exactly 2 faces adjacent to it (manifold edges), while edges with dark red outlines have either 1 face or 3+ faces adjacent to it (non-manifold).  this coloring highlights the parts of the mesh that are not closed.  you can toggle this visualization Options > Viewport Display > Target Mesh Drawing > Non-manifold Outline.

green geometry is currently selected, however the green color depends on the selected theme.  under Options > Viewport Display > Selection Theme, you can change the selection color among Green, Blue, or Orange.

the edges with a light red outline are edges that are marked as seams (Options > Pinning / Seams > Mark Selected).  edges with a yellow outline are pinned (`P` or Options > Pinning / Seams > Pin Selected).  RetopoFlow can prevent moving pinned or seam geometry (Options > Pinning / Seams > Enable Geometry Pinning and Seams Pin).

You can adjust the size of vertices and edges Options > Viewport Display > Target Mesh Drawing > Vertex Size and Edge Size.  you can also adjust the transparency of geometry with the Alpha Above and Alpha Below and the various settings under the Individual Alpha Values.

Note: we know these settings are not super easy to adjust, because we haven't spent much time on UX around configuration.  this has been a low priority item for a long time...

**Comment by vxlcoder** [comment]
digging into your issue with Contours...

if you simply double-click on an edge, that edge-loop should automatically get selected.  the behavior might be harder to predict if you're hovering a vertex, so try hovering your mouse cursor close to the middle of the edge.  does this not work consistently for you?

after the edge loop is selected, then LMB dragging a selected edge-loop should start sliding that loop similar to pressing `G`.  does this not work consistently for you?

if you're still having issues, try upgrading to RF v3.4.1 or v3.4.2beta (temp link: https://www.dropbox.com/scl/fi/7itaxnh7dmn8op7wpsp5w/RetopoFlow_v3.4.2-beta-GitHub.zip?rlkey=ztrd3stsibrjbmjtk3w8g7w41&dl=0) and try again.

if after upgrading you're still having issues, would you mind sending us your .blend file (can send to retopoflow@cgcookie.com) or creating a video capture of you trying to perform the action?  there might be something different with your settings we haven't seen that is causing problems.

thanks for helping us debug your issue!

**Comment by laptop-laptop** [comment]
I uninstalled Blender and installed the new 3.6.5 version. Tried hovering the mouse cursor close to the middle of the edge but nothing happened. Then, I uninstalled RF and installed the one you linked. This time, when I clicked on new cursor, I got an unexpected error message while trying to start RF.

**Comment by vxlcoder** [comment]
hmmm... can you copy-paste the debugging info either using deep debugging or copy-paste from the system console?  see https://docs.retopoflow.com/debugging.html#terminal--console-output for details.

**Comment by laptop-laptop** [comment]
Something very wrong is happening. Now I cannot even activate RF. Importing STL files did not give me any problem but now I can't import any STL file as doing so caused read_stl and execute error.

When do you expect the next stable version will be out? Perhaps I should wiat.

**Comment by vxlcoder** [comment]
that's very strange. the version I linked should not cause these issues, but clearly something happened. are you able to check the terminal to see what is reported there?

**Comment by laptop-laptop** [comment]
After I clicked the Import STL button, I got:

Report: Error
Python: Trackback (most recent call last):
File "C:\Program Files\Blender Foundation\Blender 3.6\3.6\scripts\addons\io_mesh_stl\_init_.py", line 126, in execute tris, tri_nors, pts = stl_utils.read_stl(path)
File "C:\Program Files\Blender Foundation\Blender 3.6\3.6\scripts\addons\io_mesh_stl\stl_utils.py", line 261, in read_stl print('Import finished in %.4f sec.' % (time.process_time() - start_time))
OSError: [WinError 6] The Handle is invalid

**Comment by vxlcoder** [comment]
did this happen after enabling deep debugging, or did it happen before?

**Comment by laptop-laptop** [comment]
If I recall correctly, it happened after I enabled deep debugging. 

I tried to uninstall RF and it gave me error. I then uninstalled Blender and reinstalled again. Got the same error after I clicked on the Import STL button. 

How come the problem in importing STL remains even after I have uninstalled Blender?

**Comment by vxlcoder** [comment]
I'm not sure.  Try uninstalling RetopoFlow (not just disable) from Blender, restart Blender, then check if Import STL still has an issue.

(i'll continue watching for replies from you until we get you back to working again...)

**Comment by laptop-laptop** [comment]
Thanks. I am trying to uninstall 3.6.5 and go back to 3.6.2 to see if the problem goes away.

How to completely uninstall and remove RetopoFlow from my PC? Got to get the import STL function working first.

**Comment by vxlcoder** [comment]
you should be able to start blender, edit .prefs, add-ons, retopoflow, unistall

**Comment by laptop-laptop** [comment]
Thanks. I uninstalled Blender 3.6.5 and reinstalled 3.6.2 which worked in the past. Unfortunately, when I clicked on Import STL, I also got error:

https://imgur.com/75d9Rg9

followed by OSError: [WinError 6] The Handle is invalid.


Then, following your suggestion, I went to edit > prefs > add-ons > retotoflow. When I clicked Remove, I got the following error 

https://imgur.com/N3ZTjzT

followed by OSError: [WinError 6] The Handle is invalid.

**Comment by laptop-laptop** [comment]
I went to Users>myuserid>AppData>Roaming>Blender Foundation>Blender>3.6>scripts>addons and deleted the RF folder. Then, decided to delete the entire Blender Foundation folder. Next, I uninstalled Blender and reinstalled Blender 3.6.2. Now, I can import stl file without problem. Not sure why the Blender Foundation folder is hidden under the Roaming folder.

So, probably RetotoFlow did not uninstall properly and somehow messed with Blender to cause error when I imported stl file. Hope developers can find a way to prevent this from happening in the future.

Perhaps I better wait for a more stable version of RF before trying again.
